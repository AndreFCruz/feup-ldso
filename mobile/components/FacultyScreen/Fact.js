import React from 'react';
import {StyleSheet} from 'react-native';
import {View, Text} from 'native-base';
import PropTypes from 'prop-types';
import facultyStyles from '../../constants/SpecificStyles';
import {widthPercentageToDP as wp} from 'react-native-responsive-screen';

class Fact extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {name, info} = this.props;
        
        let nr_display = null;
        let displayText = info;
        let matches = displayText.match(/^(\d+)\s?/); //this regex checks if displayText varible starts with a number
        if (matches != null) {
            displayText = displayText.replace(/^(\d+)\s/, ''); //this regex removes the initial number from displayText variable
            nr_display = <Text style={[styles.number, facultyStyles[name].factText]}>{matches[1]}</Text>;
        }
        return (
            <View style={[styles.fact, facultyStyles[name].factBorder]}>
                <View style={[styles.factContent, facultyStyles[name].factBackground]}>
                    {nr_display}
                    <Text style={[facultyStyles[name].factText, (matches != null ? styles.text : styles.centerText)]}>{displayText}</Text>
                </View>
            </View>
        );
    }
}

Fact.propTypes = {
    name: PropTypes.string,
    info: PropTypes.string,
};


export { Fact };


const styles = StyleSheet.create({
    fact: {
        flex:1,
        borderWidth: 3,
        margin: wp('0.65%'),
        borderRadius: 18,
        borderStyle: 'dotted',
    },
    factContent: {
        flex: 1,
        margin: wp('0.8%'),
        borderRadius: 13,
        elevation:15,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row',
    },
    number: {
        flex:1,
        textAlign:'right',
        fontSize: wp('3%'),
        paddingRight: wp('2%'),
    },  
    text: {
        flex:2,
        fontSize: wp('1.3%'),
        paddingRight: wp('2%'),
    },
    centerText: {
        flex:2,
        fontSize: wp('1.3%'),
        textAlign: 'center'
    }
});

