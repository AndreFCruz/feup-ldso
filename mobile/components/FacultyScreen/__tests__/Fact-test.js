import React from 'react';
import {Fact}  from '../Fact';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
    const wrapper = shallow(<Fact name='feup' info="240 protocolos estÃ¡gio com empresas"/>,
        {
            disableLifecycleMethods: true
        });
    return wrapper;

}

describe('Rendering ', () => {
    it('renders fact with content received', () => {
        const wrapper = setup();
        expect(wrapper.dive()).toMatchSnapshot();
    });

});