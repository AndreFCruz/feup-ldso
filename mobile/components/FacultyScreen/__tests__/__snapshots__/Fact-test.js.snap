// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering  renders fact with content received 1`] = `
<ViewNB
  style={
    Array [
      Object {},
      Object {
        "borderRadius": 18,
        "borderStyle": "dotted",
        "borderWidth": 3,
        "flex": 1,
        "margin": 5,
      },
      Object {
        "borderColor": "#9f2d20",
      },
    ]
  }
>
  <Styled(ViewNB)
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 13,
          "elevation": 15,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 6,
        },
        Object {
          "backgroundColor": "#9f2d20",
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 22.5,
            "paddingRight": 15,
            "textAlign": "right",
          },
          Object {
            "color": "#ffffff",
          },
        ]
      }
    >
      240
    </Styled(Text)>
    <Styled(Text)
      style={
        Array [
          Object {
            "color": "#ffffff",
          },
          Object {
            "flex": 2,
            "fontSize": 10,
            "paddingRight": 15,
          },
        ]
      }
    >
      protocolos est√°gio com empresas
    </Styled(Text)>
  </Styled(ViewNB)>
</ViewNB>
`;
